kube-controllers/pkg/controllers/node/ipam.go

```diff
@@ -953,6 +1141,9 @@ func (c *ipamController) garbageCollectIPs() error {
 		delete(c.allocationsByNode[a.node()], id)
 		if len(c.allocationsByNode[a.node()]) == 0 {
 			delete(c.allocationsByNode, a.node())
+			clearNodeFromPoolCounters(a.node())
+		} else {
+			c.incrementReclamationMetric(id, a.node())
```

Comment:
The idea was to prevent a case where a counter is published that will never get cleaned up, since the current implementation cleans up a counter when its node has no more allocations.

Given the valid suggestion below on changing the cleanup trigger, this conditional will go away